### PR TITLE
Revert "DEV: Fixed Un-responsive live-preview in gitpod."

### PR DESCRIPTION
**Pull request by bhavukkalra**
Reverts numpy/numpy#21250.

`# NOTE: breathe 4.33.0 collides with sphinx.ext.graphviz`

@rossbar [Commit](https://github.com/numpy/numpy/commit/41cf10dcb6ab46064d73f2c054e618c76c3cfabc) Ran with Skip-CI seems to have broken the builds.

This is a Revert PR.

### CODE REVIEW

**Comment by rossbar** [comment]
I don't think this is necessary - what specifically is broken? I can `conda env create -f environment.yml` just fine.

Is it the gitpod that's not working? I'm not sure how long we have to wait before the image updates...

**Comment by bhavukkalra** [comment]
It might be the case that Gitpod might be taking sometime to pick up the latest `environment.yml`(I am not quiet sure on this). As there is an ambiguity on the environment that I am working on.
1) when I did opened up an Instance it still seems to reflect breathe version as `4.33.0`. Through which I though that `environment.yml` is not picked up

1) But `pip install esbonio` seems to be a command that is being ran. which contradicts with the above observation as both changes were made in a single commit.

Below are the Contradictory logs(Highlighted)
contradiction_logs

**Comment by rossbar** [comment]
> when I did opened up an Instance it still seems to reflect breathe version as 4.33.0. Through which I though that environment.yml is not picked up

I see that in the gitpod as well, though I suspect it is because the environment is not created from the `environment.yml` each time a gitpod instance is launched. For example, the `gitpod.yml` [only activates the environment](https://github.com/numpy/numpy/blob/41cf10dcb6ab46064d73f2c054e618c76c3cfabc/.gitpod.yml#L14). 

> But pip install esbonio seems to be a command that is being ran. which contradicts with the above observation as both changes were made in a single commit.

This is actually *not* working as expected - I had hoped that this would install `esbonio` in the conda environment, but if you look closely at the logs in the gitpod terminal, it's actually installing it in a `/home/gitpod/.local` location which is not on PYTHONPATH, meaning that the installed package is not picked up at all. That's why there's a second, separate download step during the sphinx build which shouldn't be necessary if it were already installed.

Long story short, the gitpod setup/config is more complicated than I originally realized. Any ideas @melissawm @trallard ? Feel free to merge this reversion if it helps (though it just puts gitpod back into a different broken state).

**Comment by bhavukkalra** [comment]
Agreed. There is a lot of stuff going on like creating a local cache directory, A different workspace being created altogether for every gitpod instance Initiated. I will need to dig more deep into its working. To find out what are the constraints that we are dealing with and How to get the above changes to work as they are supposed to. For now, We should keep this revert PR around as a reminder to dig deep, until I figure something out.

**Comment by bhavukkalra** [comment]
I think I might have found why breathe is not getting picked up but `pip install esbonio` is.
See, Gitpod takes in `numpy-dev:latest` image from numpy dockerhub as a base. And it already has the environment created i.e `numpy-dev`.
On gitpod we just activate it via `conda activate numpy-dev`.

At this time I think, the deployed docker Image still has the old `environment.yml` until a new one is deployed.(We can do it, but needs discussion).

Other commands like `pip install esbonio` are ran on top of the Base Docker Hub(already deployed) container.

This is the reasoning I came up when I saw the observation above that `breathe!=4.33.0` wasn't getting picked up but `pip install esbonio was`

**Comment by trallard** [comment]
That workflow is correct

1. Whenever a change to `environment.yml` is made in `main` the GH action workflow will rebuild the image and push to DockerHub
2. The gitpod Docker image is rebuilt on every push to main - to keep the NumPy version in sync wth the head -> this is then used for the Gitpod prebuild
3. Whenever one starts a Gitpod instance then the init steps in `.gitpod.yml` get executed. 

I can have a look at this today/tomorrow.


BTW the `numpy/numpy-dev` image was rebuilt 9 days ago which is when https://github.com/numpy/numpy/pull/21250 got merged

trallard_mictlan___Documents_github_patches_numpy
